test/server/guarddog_impl_test.cc

```diff
@@ -194,6 +197,19 @@ class GuardDogMissTest : public GuardDogTestBase {
 protected:
   GuardDogMissTest() : config_miss_(500, 1000, 0, 0), config_mega_(1000, 500, 0, 0) {}
 
+  void checkMiss(uint64_t count) {
+    EXPECT_EQ(count, TestUtility::findCounter(stats_store_, "server.watchdog_miss")->value());
```

Comment:
One thing that might be annoying is that if this fails -- say with a flake, you won't know which checkMiss call it was that failed. Maybe pass a string descriptor in so that you could then write:
```
EXPECT_EQ(count, TestUtility:...) << descriptor;
```
and then you could pass in unique strings in the calls to checkMiss and checkMegaMiss.